src/JaxkDev/Vehicles/Handlers/EventHandler.php

```diff
@@ -87,17 +90,43 @@ public function onPlayerTeleportEvent(EntityTeleportEvent $event): void{
 	 * Some interruption by MultiWorld
 	 */
 	public function onEntityDamageEvent(EntityDamageByEntityEvent $event): void{
-		if($event->getEntity() instanceof Vehicle){
-			$event->setCancelled(); //stops the ability to 'kill' a object/vehicle. (In long future, add vehicle condition *shrug*
-			if(!($event->getDamager() instanceof Player)) return;
-			/** @var Player $attacker */
-			$attacker = $event->getDamager();
-			/** @var Vehicle $entity */
-			$entity = $event->getEntity();
+		if($event->getEntity() instanceof Vehicle) {
+            $event->setCancelled(); //stops the ability to 'kill' a object/vehicle. (In long future, add vehicle condition *shrug*
+            if (!($event->getDamager() instanceof Player)) return;
+            /** @var Player $attacker */
+            $attacker = $event->getDamager();
+            /** @var Vehicle $entity */
+            $entity = $event->getEntity();
+            // item
+            $item = Item::get(Item::NETHER_STAR, 0, 1);
+            $item->setCustomName(C::BOLD . C::RED . "Car");
+            $item->setNamedTagEntry(new IntTag("car", 1));
+            $item->getNamedTagEntry("car");
+
+            ###
+            // remove by interact
+            if ($entity instanceof Vehicle) {
+                if (!$entity->isVehicleEmpty()) {
+                    $attacker->sendMessage(Main::$prefix . C::RED . "You cannot remove a vehicle with players in it.");
+                } else {
+                    $entity->close();
+                    $attacker->sendMessage(Main::$prefix . "'" . $entity->getVehicleName() . "' has been removed.");
+                    if (($inv = $attacker->getInventory()) == null) {
+                        $attacker->getLevel()->dropItem(Position::fromObject($attacker->add(0.5, 0.75, 0.5)), ($item));
+                    } else {
+                        $attacker->getInventory()->addItem($item);
+                    }
+                    $item->setCount(1);
+                }
+                return;
```

Comment:
This would stop everything after this including logic for driving, events, and passengers getting in.